mautic/mautic
# Select with Mautic tokens for GrapesJS text editor

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | No
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | [ ]
| Related user documentation PR URL      | []
| Related developer documentation PR URL | []
| Issue(s) addressed                     | Connected with #10966 #11344 

#### Description:

GrapesJS builder didn't have option to insert Mautic tokens ([https://docs.mautic.org/en/setup/variables](https://docs.mautic.org/en/setup/variables)) in the body of an email with help of a drop-down list with hints, just like in the old builder. 
![token_prev_4](https://user-images.githubusercontent.com/102536220/193092028-180a37d2-6f99-4029-94b8-1d78d33c9259.jpg)

I added a dropdown with tokens as a CKEditor plugin loaded only in the GrapesJS builder. I didn't implement the option to display hints after typing {. In the drop-down variables are divided into few groups: Assets, Contact fields, Company fields, DWC, Page links, Custom tokens and are inserted into the text in the same way as in the old builder.
![token_prev_1](https://user-images.githubusercontent.com/102536220/193093071-b4d7d68a-cd55-4571-a85f-0bda0ce19d58.jpg)


#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Enable the GrapesJS plugin.
3. Create a new e-mail or open an existing e-mail for editing.
4. Double-click any E-Mail text to display the editor toolbar.
5. The editor toolbar should contain the "Token" drop-down list. Click an option from the list to insert it into the text.


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11525"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
